### PR TITLE
Add a flight control board for BAYCKRC

### PR DESCRIPTION
**Pull request by BAYCKRC**
This commit adds the BAYCK_DOLPHIN_F435 configuration file, which includes various settings and pin mappings for the target MCU AT32F435G. The configuration enables the use of accelerometer (ACC), gyroscope (GYRO), barometer (BARO), magnetometer (MAG), and MAX7456 OSD. It also defines pin assignments for motors, UARTs, LEDs, SPI interfaces, ADC pins, and timers. Additionally, it sets default values for blackbox device, current meter source and scale, voltage meter source and scale, and beeper inversion.

### CODE REVIEW

File: configs/BAYCK_DOLPHIN_F435/config.h [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define FC_TARGET_MCU     AT32F435G
+
+#define BOARD_NAME        BAYCK DOLPHIN
+#define MANUFACTURER_ID   BAYCKRC
```
Comment (nerdCopter) [inline configs/BAYCK_DOLPHIN_F435/config.h:27]: manufacturer id is 4 characters only.

Comment (haslinghuis) [reply configs/BAYCK_DOLPHIN_F435/config.h:27]: Please mention website url for the product.

Comment (BAYCKRC) [reply configs/BAYCK_DOLPHIN_F435/config.h:27]: We currently have a website, but we haven't had time to build it yet. It is expected to be completed within 3 months. Also, does the website need to be written to the config. h file?

Comment (haslinghuis) [reply configs/BAYCK_DOLPHIN_F435/config.h]: No there is no need to include url in the config. We need to have some info on the board itself. 
We have a wiki link for each board pointing to documentation (like board specification, layout and wire diagram)

File: configs/BAYCK_DOLPHIN_F435/config.h [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    TIMER_PIN_MAP( 6, PB6 , 1,  13) \
+    TIMER_PIN_MAP( 7, PB8 , 2, -1) \
+    TIMER_PIN_MAP( 8, PC6 , 2,  11) \
+    TIMER_PIN_MAP( 9, PC7 , 2,  10) \
+    TIMER_PIN_MAP(10, PA9 , 1,  5) \
+    TIMER_PIN_MAP(11, PA10, 1,  4) \
+    TIMER_PIN_MAP(12, PA1 , 1,  12) \
+    TIMER_PIN_MAP(13, PA2 , 1, -1)
+
+#define ADC1_DMA_OPT        11
```
Comment (ot0tot) [inline configs/BAYCK_DOLPHIN_F435/config.h:105]: Should this be 
`#define ADC2_DMA_OPT        11` 
since you are using ADC2?
`#define ADC_INSTANCE ADC2`

**Comment by BAYCKRC** [comment]
Good!
&nbsp;I understand the wiki link you mentioned and will contact you immediately after our website construction is completed




---Original---
From: "Mark ***@***.***&gt;
Date: Wed, Sep 13, 2023 21:25 PM
To: ***@***.***&gt;;
Cc: "BAYCK ***@***.******@***.***&gt;;
Subject: Re: [betaflight/config] Add a flight control board for BAYCKRC (PR#170)




 
@haslinghuis commented on this pull request.
 
 
In configs/BAYCK_DOLPHIN_F435/config.h:
 &gt; + * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. + * + * See the GNU General Public License for more details. + * + * You should have received a copy of the GNU General Public + * License along with this software. + * + * If not, see <http://www.gnu.org/licenses/&gt;. + */ + +#pragma once + +#define FC_TARGET_MCU     AT32F435G + +#define BOARD_NAME        BAYCK DOLPHIN +#define MANUFACTURER_ID   BAYCKRC  
No there is no need to include url in the config. We need to have some info on the board itself.
 We have a wiki link for each board pointing to documentation (like board specification, layout and wire diagram)

File: configs/BAYCK_DOLPHIN_F435/config.h (hunk 2 above)
Comment (BAYCKRC) [reply configs/BAYCK_DOLPHIN_F435/config.h:105]: No, my terminal shows adc1。Here is the log information：

# 
# Building AutoComplete Cache ... Done!
# 
# dma
dma ADC 1 11
# ADC 1: DMA2 Channel 0 Channel 5
dma pin B00 7
# pin B00: DMA2 Channel 0 Channel 63
dma pin B01 2
# pin B01: DMA1 Channel 0 Channel 64
dma pin A03 6
# pin A03: DMA1 Channel 0 Channel 59
dma pin B05 3
# pin B05: DMA1 Channel 0 Channel 62
dma pin C08 9
# pin C08: DMA2 Channel 0 Channel 51
dma pin C09 8
# pin C09: DMA2 Channel 0 Channel 52
dma pin B06 13
# pin B06: DMA2 Channel 0 Channel 67
dma pin C06 11
# pin C06: DMA2 Channel 0 Channel 49
dma pin C07 10
# pin C07: DMA2 Channel 0 Channel 50
dma pin A09 5
# pin A09: DMA1 Channel 0 Channel 43
dma pin A10 4
# pin A10: DMA1 Channel 0 Channel 44
dma pin A01 12
# pin A01: DMA2 Channel 0 Channel 57

# dma show

Currently active DMA:
--------------------
DMA1 Channel 1: DSHOT_BITBANG 2
DMA1 Channel 2: DSHOT_BITBANG 1
DMA1 Channel 3: SPI_SDO 1
DMA1 Channel 4: SPI_SDI 1
DMA1 Channel 5: SPI_SDO 2
DMA1 Channel 6: SPI_SDI 2
DMA1 Channel 7: SPI_SDO 3
DMA2 Channel 1: SPI_SDI 3
DMA2 Channel 2: FREE
DMA2 Channel 3: FREE
DMA2 Channel 4: FREE
DMA2 Channel 5: ADC 1
DMA2 Channel 6: FREE
DMA2 Channel 7: LED_STRIP

Comment (ot0tot) [reply configs/BAYCK_DOLPHIN_F435/config.h:105]: Yes, this is the problem. You are assigning DMA to ADC1, but tell Betaflight to use ADC2. 

You need to either assign DMA to ADC2, or tell Betaflight to use ADC1.

File: configs/BAYCK_DOLPHIN_F435/config.h [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+#define MAG_I2C_INSTANCE (I2CDEV_1)
+#define BARO_SPI_INSTANCE SPI3
+#define USE_ADC
+#define ADC_INSTANCE ADC2
```
Comment (ot0tot) [inline configs/BAYCK_DOLPHIN_F435/config.h]: If you are using ADC1, then this should be
`#define ADC_INSTANCE ADC1`

Comment (BAYCKRC) [reply configs/BAYCK_DOLPHIN_F435/config.h]: Now refresh, I'm sure I've made the changes

File: configs/BAYCK_DOLPHIN_F435/config.h [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+#define USE_GYRO
+#define USE_GYRO_SPI_MPU6000
+#define USE_GYRO_SPI_MPU6500
+#define USE_ACC_SPI_ICM42688P
+
+#define USE_ACC
+#define USE_ACC_SPI_MPU6000
+#define USE_ACC_SPI_MPU6500
+#define USE_GYRO_SPI_ICM42688P
```
Comment (ot0tot) [inline configs/BAYCK_DOLPHIN_F435/config.h:37]: Should be `#define USE_ACC_SPI_ICM42688P`

File: configs/BAYCK_DOLPHIN_F435/config.h [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+#define FC_TARGET_MCU     AT32F435G
+
+#define BOARD_NAME        BAYCK_DOLPHIN
+#define MANUFACTURER_ID   BKRC
+
+#define USE_GYRO
+#define USE_GYRO_SPI_MPU6000
+#define USE_GYRO_SPI_MPU6500
+#define USE_ACC_SPI_ICM42688P
```
Comment (ot0tot) [inline configs/BAYCK_DOLPHIN_F435/config.h:32]: Should be `#define USE_GYRO_SPI_ICM42688P`

**Comment by BAYCKRC** [comment]
Thank you very much for your patient and attentive guidance

**Comment by BAYCKRC** [comment]
Thank you very much for your patient and attentive guidance

**Review by haslinghuis — APPROVED** [review]

**Comment by nerdCopter** [comment]
regarding the long term use of,
`#define BOARD_NAME        BAYCK_DOLPHIN`

some things to consider:
- will there be more boards in the future?
- will all boards need `BAYCK_` in the name?
- maybe just `DOLPHIN`
- or maybe `DOLPHINF435` 

i dont know, personally... just something to consider for long-term target lists. 
thoughts @haslinghuis ?

**Comment by haslinghuis** [comment]
Yes like DOLPHINF435

**Comment by BAYCKRC** [comment]
At least three more boards need to be added in the future. All board names start with BAYCK_. The current board is our first product, and we will name it DOLPHIN. The board to be released later will be BAYCK_ XXXXX, so # define BOARD_ NAME, we only take BAYCK_ DOLLIN.



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 13, 2023 23:09 PM
To: ***@***.***&gt;;
Cc: "BAYCK ***@***.***&gt;;"State ***@***.***&gt;;
Subject: Re: [betaflight/config] Add a flight control board for BAYCKRC (PR#170)




 
regarding the long term use of,
 #define BOARD_NAME        BAYCK_DOLPHIN
 
some things to consider:
  
will there be more boards in the future?
 
will all boards need BAYCK_ in the name?
 
maybe just DOLPHIN
 
or maybe DOLPHINF435
  
i dont know, personally... just something to consider for long-term target lists.
 thoughts @haslinghuis ?

**Review by nerdCopter — APPROVED** [review]

**Comment by BAYCKRC** [comment]
We don't like this name DOLPHINF435。
In the Betaflight Configurator, the current flight control model already displays BKRC/BAYCK_ DOLPHIN (AT32F435G), so there is no need to add the F435 suffix



---Original---
From: "Mark ***@***.***&gt;
Date: Wed, Sep 13, 2023 23:10 PM
To: ***@***.***&gt;;
Cc: "BAYCK ***@***.***&gt;;"State ***@***.***&gt;;
Subject: Re: [betaflight/config] Add a flight control board for BAYCKRC (PR#170)




 
Yes like DOLPHINF435

**Comment by BAYCKRC** [comment]
Thank you very much for your hard work and dedication!
&nbsp;When is the online server compilation expected to be available in the Betafligt Configurator?



---Original---
From: "Mark ***@***.***&gt;
Date: Wed, Sep 13, 2023 23:41 PM
To: ***@***.***&gt;;
Cc: "BAYCK ***@***.***&gt;;"State ***@***.***&gt;;
Subject: Re: [betaflight/config] Add a flight control board for BAYCKRC (PR#170)




 
Merged #170 into master.

**Comment by haslinghuis** [comment]
image

**Comment by BAYCKRC** [comment]
I checked the firmware name 30 minutes ago, but during the compilation process, there was a prompt to save and the compilation command has been terminated. I need to rest and sleep now. I'll try again in a few hours



---Original---
From: "Mark ***@***.***&gt;
Date: Thu, Sep 14, 2023 00:25 AM
To: ***@***.***&gt;;
Cc: "BAYCK ***@***.***&gt;;"State ***@***.***&gt;;
Subject: Re: [betaflight/config] Add a flight control board for BAYCKRC (PR#170)

**Comment by haslinghuis** [comment]
Enabled show release candidates and expert mode and select development mode

**Comment by BAYCKRC** [comment]
I have found the problem that cannot be compiled. My board name is BYCK_ DOLPHIN, but config. h is in configs/BACK_ DOLPHIN_ F435/config. h. So, BAYCK_ DOLPHIN_ The F435 file directory is the problem.


Online compilation is the process of compiling the BYCK in the BOARD NAME_ DOLPHIN is passed to 'make' to form 'make BACK'_ DOLPHIN ', but in order to successfully compile,' make BACK 'is required_ DOLPHIN_ F435 '. So, should I modify the directory 'BAYCK'_ DOLPHIN_ F435 is' BAYCK '_ DOLPHIN??


Alternatively, you can point the index of Betaflight Configurator to the directory 'BACK'_ DOLPHIN_ F435???


The following is the compilation log:



***** Commencing Build ***** VERSION:   4.5.0-zulu COMMIT:    99be07729963e4b5e5cb540555450bda28bd2781 CONTAINER: master BOARD:     BAYCK_DOLPHIN/AT32F435G STARTED:   2023-09-14 04:47:31Z Want to run this build locally? NOTE: you will need docker installed. See https://github.com/betaflight/cloudbuild for instructions. &gt; docker run betaflight/cloudbuild:master /build/build.sh 99be07729963e4b5e5cb540555450bda28bd2781 BAYCK_DOLPHIN "-D'BUILD_KEY=4497c8df8baee22521f190503305ce8c' -D'RELEASE_NAME=4.5.0-zulu' -DCLOUD_BUILD -DUSE_DSHOT -DUSE_GPS -DUSE_GPS_PLUS_CODES -DUSE_LED_STRIP -DUSE_OSD -DUSE_OSD_HD -DUSE_OSD_SD -DUSE_PINIO -DUSE_SERIALRX -DUSE_SERIALRX_CRSF -DUSE_TELEMETRY -DUSE_TELEMETRY_CRSF -DUSE_VTX"  -- or (without docker) --  &gt; make BAYCK_DOLPHIN EXTRA_FLAGS="-D'BUILD_KEY=4497c8df8baee22521f190503305ce8c' -D'RELEASE_NAME=4.5.0-zulu' -DCLOUD_BUILD -DUSE_DSHOT -DUSE_GPS -DUSE_GPS_PLUS_CODES -DUSE_LED_STRIP -DUSE_OSD -DUSE_OSD_HD -DUSE_OSD_SD -DUSE_PINIO -DUSE_SERIALRX -DUSE_SERIALRX_CRSF -DUSE_TELEMETRY -DUSE_TELEMETRY_CRSF -DUSE_VTX" ***** Docker execution log: ***** Pulling from Github (99be07729963e4b5e5cb540555450bda28bd2781): From https://github.com/betaflight/betaflight  * [new ref]             99be07729963e4b5e5cb540555450bda28bd2781 -&gt; CLOUD_HEAD Checking out the branch -&gt; CLOUD_HEAD (for 99be07729963e4b5e5cb540555450bda28bd2781) Switched to branch 'CLOUD_HEAD' Commencing the build (make BAYCK_DOLPHIN): git -C ./src/config pull origin From https://github.com/betaflight/config    ae5ad55..6f3a372  master     -&gt; origin/master Updating ae5ad55..6f3a372 Fast-forward  Manufacturers.md                                |  95 ++++++++++++++++++  configs/BAYCK_DOLPHIN_F435/config.h             | 122 ++++++++++++++++++++++++  configs/{GEPRCF722AIO =&gt; HGLRCF722AIO}/config.h |   0  3 files changed, 217 insertions(+)  create mode 100644 Manufacturers.md  create mode 100644 configs/BAYCK_DOLPHIN_F435/config.h  rename configs/{GEPRCF722AIO =&gt; HGLRCF722AIO}/config.h (100%) make: *** No rule to make target 'BAYCK_DOLPHIN'.  Stop. make failed with 2  ***** Completed Build ***** Build time: 2023-09-14 04:47:38Z, time taken: 7.42 Submitted:  2023-09-14 04:47:29Z, wait time: 8.72




1003479366
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "betaflight/config"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年9月14日(星期四) 凌晨0:36
***@***.***&gt;;
***@***.***&gt;;"State ***@***.***&gt;;
主题:&nbsp;Re: [betaflight/config] Add a flight control board for BAYCKRC (PR #170)





 
Enabled show release candidates and expert mode and select development mode

**Comment by nerdCopter** [comment]
https://github.com/betaflight/config/pull/172 merged.
tested locally: `Building target config BAYCK_DOLPHIN succeeded.`

**Pull request merged**